### ISSUE TITLE
Replacing zero by empty array blows up runtime and memory in trivial GROUP BY 

### ISSUE THREAD

**Issue by soerenwolfers** [duckdb/duckdb#10989]
### What happens?

Replacing zero by empty array blows up runtime and memory in trivial GROUP BY.

### To Reproduce

```sql
SELECT 
first(value)
FROM (SELECT *, [] AS value FROM range(1_000_000))
GROUP BY range
```
takes 10s and 50GB while 
```sql
SELECT 
first(value)
FROM (SELECT *, 0 AS value FROM range(1_000_000))
GROUP BY range
```
is instant. 

It's not a great example because you could imagine that in both cases the optimizer could optimize away doing anything, but I don't think that's what's happening in either case as 10s and 50GB is way too much for even an unoptimized iteration through 16MB of data. I'm posting this after a long time of boiling down performance problems in a real problem. 

All the time is spent in the hash group by according to EXPLAIN ANALYZE.

### OS:

Linux

### DuckDB Version:

0.10.1.dev784

### DuckDB Client:

Python

### Full Name:

Soeren Wolfers

### Affiliation:

G-Research

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by soerenwolfers** [comment]
Still an issue on 1.0.0 (got worse in fact)

**Comment by soerenwolfers** [comment]
Fixed in 1.1
